chef/cookbooks/crowbar/templates/default/crowbar-delete-pacemaker-resources.sh.erb

```diff
@@ -0,0 +1,36 @@
+#!/bin/bash
+#
+# Old pacemaker resources must be deleted in the old setup before they get created from upgraded node
+
+LOGFILE=/var/log/crowbar/node-upgrade.log
+mkdir -p "`dirname "$LOGFILE"`"
+exec >>"$LOGFILE" 2>&1
+
+echo "Executing $BASH_SOURCE"
+
+set -x
+
+<% if @use_ha %>
+
+echo "Stopping pacemaker resources..."
+
+
+for type in clone group ms primitive; do
+    for resource in $(crm configure show | awk "\$1 == \"$type\" && ! (\$2 ~ /drbd|stonith|vip-/) {print \$2}"); do
+        crm --wait resource stop $resource
+    done
+done
+
+echo "Deleting pacemaker resources..."
+
+for type in location clone group ms primitive; do
```

Comment:
What about `order` and `colocation`?

Also, there should be a comment explaining why it's crucial that services are stopped before deletion, in order to make it safe to remove constraints in separate singleton transactions before removing the corresponding services they refer to.